homeassistant/components/tuya/base.py

```diff
@@ -86,6 +86,12 @@ def from_json(cls, dpcode: DPCode, data: str) -> IntegerTypeData | None:
             unit=parsed.get("unit"),
             type=parsed.get("type"),
         )
+        if (dpcode == DPCode.TEMP_UPPER or dpcode == DPCode.TEMP_UPPER_F) or dpcode == DPCode.TEMP_CURRENT or dpcode == DPCode.TEMP_CURRENT_F):
```

Comment:
as already stated above from the documentation: step can only be "0" for current temperature. this denotes rounding to integer. whereas step "1" for current temperature stands for decimal result.

Thus for all other values step must be > 0

Reply:
This logic is not just for temperatures, but applies to all datapoints.
We should not have datapoint specific exceptions here.

Reply:
https://developer.tuya.com/en/docs/iot/shift-temperature-scale-follow-the-setting-of-app-account-center?id=Ka9qo7so58efq#title-8-Example%201%3A%20set%20the%20scale%20to%200

https://developer.tuya.com/en/docs/iot/shift-temperature-scale-follow-the-setting-of-app-account-center?id=Ka9qo7so58efq#title-8-Example%201%3A%20set%20the%20scale%20to%201

Reply:
Yes, I read that, but those are 2 different problems in HA.

Reply:
Since step is an integer in the json data this should work :-)
Then the only exception in scaling is the step itself, because it is not required to be multiplied by itself...

Thus to summarize the result:

* Constructing the class from json
step = max(1, stepFromJsom)

* Scaling the step
scaled_step = step / (10^scale)

* Scaling values
scaled_value = value * step / (10^scale)

Correct?